storm-core/test/jvm/org/apache/storm/tuple/FieldsTest.java

```diff
@@ -0,0 +1,122 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.storm.tuple;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class FieldsTest {
+
+    @Test
+    public void fieldsConstructorDoesNotThrowWithValidArgsTest() {
+        Assert.assertEquals(new Fields("foo", "bar").size(), 2);
+        Assert.assertEquals(new Fields(new String[] {"foo", "bar"}).size(), 2);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void duplicateFieldsNotAllowedWhenConstructingWithVarArgsTest() {
+        new Fields("foo", "bar", "foo");
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void duplicateFieldsNotAllowedTestWhenConstructingFromListTest() {
+        new Fields(new String[] {"foo", "bar", "foo"});
+    }
+
+    private Fields getFields() {
+        return new Fields("foo", "bar");
```

Comment:
There are a lot of places where I would prefer to create the Fields instance once and then manipulate it, instead of creating a new one each time.  Also I find it cleaner to see the class being instantiated right by where they are tested.